cmd/juju/model/setcredential.go

```diff
@@ -0,0 +1,188 @@
+type ModelCredentialAPI interface {
+	Close() error
+	ChangeModelCredential(model names.ModelTag, credential names.CloudCredentialTag) error
+}
+
+// CloudAPI defines methods used to detemine if cloud credential exists on the controller.
+type CloudAPI interface {
+	Close() error
+	UserCredentials(names.UserTag, names.CloudTag) ([]names.CloudCredentialTag, error)
+	AddCredential(tag string, credential cloud.Credential) error
+}
+
+// modelCredentialCommand allows to change, replace a cloud credential for a model.
+type modelCredentialCommand struct {
+	modelcmd.ModelCommandBase
+
+	cloud      string
+	credential string
+
+	newAPIRootFunc            func() (base.APICallCloser, error)
+	newModelCredentialAPIFunc func(base.APICallCloser) ModelCredentialAPI
+	newCloudAPIFunc           func(base.APICallCloser) CloudAPI
+}
+
+func NewModelCredentialCommand() cmd.Command {
+	command := &modelCredentialCommand{
+		newModelCredentialAPIFunc: func(root base.APICallCloser) ModelCredentialAPI {
+			return modelmanager.NewClient(root)
+		},
+		newCloudAPIFunc: func(root base.APICallCloser) CloudAPI {
+			return cloudapi.NewClient(root)
+		},
+	}
+	command.newAPIRootFunc = func() (base.APICallCloser, error) {
+		return command.NewControllerAPIRoot()
+	}
+	return modelcmd.Wrap(command)
+}
+
+// Info implements Command.Info.
+func (c *modelCredentialCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "set-credential",
+		Args:    "<cloud name> <credential name>",
+		Purpose: "Change cloud credential for the current or specified model.",
+		Doc:     modelCredentialDoc,
+	}
+}
+
+// SetFlags implements Command.SetFlags.
+func (c *modelCredentialCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+}
+
+// Init implements Command.Init.
+func (c *modelCredentialCommand) Init(args []string) error {
+	if len(args) != 2 {
+		return errors.Errorf("Usage: juju set-credential [options] <cloud name> <credential name>")
+	}
+	if !names.IsValidCloud(args[0]) {
+		return errors.NotValidf("cloud name %q", args[0])
+	}
+	if !names.IsValidCloudCredentialName(args[1]) {
+		return errors.NotValidf("cloud credential name %q", args[1])
+	}
+	c.cloud = args[0]
+	c.credential = args[1]
+	return nil
+}
+
+// Run implements Command.Run.
+func (c *modelCredentialCommand) Run(ctx *cmd.Context) error {
+	fail := func(e error) error {
+		ctx.Infof("Failed to change model credential: %v", e)
+		return e
+	}
+
+	root, err := c.newAPIRootFunc()
+	if err != nil {
+		return fail(errors.Annotate(err, "opening API connection"))
+	}
+	defer root.Close()
+
+	accountDetails, err := c.CurrentAccountDetails()
+	if err != nil {
+		return fail(errors.Annotate(err, "getting current account"))
+	}
+	userTag := names.NewUserTag(accountDetails.User)
+	cloudTag := names.NewCloudTag(c.cloud)
+	credentialTag, err := common.ResolveCloudCredentialTag(userTag, cloudTag, c.credential)
+	if err != nil {
+		return fail(errors.Annotate(err, "resolving credential tag"))
```

Comment:
The error shouldn't mention "tag" (they're an implementation detail of the API) - I think "resolving credential" would be fine.

Reply:
Copy from elsewhere but true, I'll change this instance.